force-app/main/default/classes/EmailServiceInstaller.cls

```diff
@@ -88,4 +88,66 @@ public with sharing class EmailServiceInstaller {
         Http h = new Http();
         HttpResponse res = h.send(req);
     }
+
+    public static void deleteEmailServiceFunction(EmailServicesFunction emailServiceFunction)
+    {
+        String baseUrl = URL.getOrgDomainUrl().toExternalForm() + '/services/Soap/u/48.0';
+
+        HTTPRequest req = new HTTPRequest();
+        req.setEndpoint(baseUrl);
+        req.setMethod('POST');
+        req.setHeader('SOAPAction', '""');
+        req.setHeader('Content-Type', 'text/xml');
+        req.setBody(
+            '<?xml version="1.0" encoding="utf-8"?>' +
+              '<env:Envelope xmlns:env="http://schemas.xmlsoap.org/soap/envelope/" xmlns:urn="urn:partner.soap.sforce.com">' +
+                '<env:Header>' +
+                    '<urn:SessionHeader>' +
+                        '<urn:sessionId>'+UserInfo.getSessionId()+'</urn:sessionId>' +
+                    '</urn:SessionHeader>' +
+                '</env:Header>' +
+                '<env:Body>' +
+                    '<urn:delete>' +
+                        '<urn:ids>' +
+                            emailServiceFunction.Id +
+                        '</urn:ids>' +
+                    '</urn:delete>' +
+                '</env:Body>' +
+            '</env:Envelope>'
+        );
```

Comment:
Is there a better way to do this? This might be ok, but I was a little uncomfortable with building XML as a string object in a function like this...

Reply:
You could use `XmlStreamWriter` but it would be more verbose and harder to read at a glance. It would have more advantages if the XML were more dynamic or more composable. These are mostly just static templates.

Reply:
Can probably use `StringBuilder` here if we want

Reply:
Apex doesn't have `StringBuilder` built in, but it could be implemented.

Reply:
@waltjones I didn't know that, my bad